File: server/src/main/java/com/grpc/empuje_comunitario/domain/user/EmailValidator.java
package com.grpc.empuje_comunitario.domain.user;

public class EmailValidator implements FieldValidator{
    @Override
    public boolean validate(String value) {
        return value != null && value.matches("^[^@\\s]+@[^@\\s]+\\.[^@\\s]+$");
    }
}

File: server/src/main/java/com/grpc/empuje_comunitario/domain/user/EmailValidator.kt
package com.grpc.empuje_comunitario.domain.user

class EmailValidator : FieldValidator {
    override fun validate(value: String?): Boolean {
        return value != null && Regex("^[^@\\s]+@[^@\\s]+\\.[^@\\s]+$").matches(value)
    }
}